cli/js/lib.deno.ns.d.ts

```diff
@@ -33,69 +33,67 @@ declare namespace Deno {
    * when `Deno.runTests` is used */
   export function test(name: string, fn: TestFunction): void;
 
-  enum TestStatus {
-    Passed = "passed",
-    Failed = "failed",
-    Skipped = "skipped"
-  }
-
-  interface TestResult {
-    name: string;
-    status: TestStatus;
-    duration?: number;
-    error?: Error;
-  }
-
-  interface TestStats {
-    filtered: number;
-    ignored: number;
-    measured: number;
-    passed: number;
-    failed: number;
-  }
-
-  export enum TestEvent {
+  enum TestEvent {
     Start = "start",
     TestStart = "testStart",
     TestEnd = "testEnd",
     End = "end"
   }
 
-  interface TestEventStart {
-    kind: TestEvent.Start;
-    tests: number;
-  }
-
-  interface TestEventTestStart {
-    kind: TestEvent.TestStart;
-    name: string;
-  }
-
-  interface TestEventTestEnd {
-    kind: TestEvent.TestEnd;
-    result: TestResult;
-  }
-
-  interface TestEventEnd {
-    kind: TestEvent.End;
-    stats: TestStats;
-    duration: number;
-    results: TestResult[];
-  }
-
-  interface TestReporter {
-    start(event: TestEventStart): Promise<void>;
-    testStart(msg: TestEventTestStart): Promise<void>;
-    testEnd(msg: TestEventTestEnd): Promise<void>;
-    end(event: TestEventEnd): Promise<void>;
-  }
-
-  export class ConsoleTestReporter implements TestReporter {
-    constructor();
-    start(event: TestEventStart): Promise<void>;
-    testStart(msg: TestEventTestStart): Promise<void>;
-    testEnd(msg: TestEventTestEnd): Promise<void>;
-    end(event: TestEventEnd): Promise<void>;
+  export namespace tests {
+    export enum Status {
+      Passed = "passed",
+      Failed = "failed",
+      Skipped = "skipped"
+    }
+
+    export interface Result {
+      name: string;
+      status: Status;
+      duration: number;
+      error?: Error;
+    }
+
+    export interface StartMessage {
+      tests: TestDefinition[];
```

Comment:
rollback, no reason to send (non-serializable) test definition

Reply:
The custom `Reporter` API, through which this is public, doesn't need it to be serializable. I think this is just required in `unit_test_runner.ts`/`test_util.ts` which have their own layer of communication.

Reply:
Is this okay? My point here is that the serializability constraint is only applicable to `SocketReporter` and not `Reporter`s in general.